# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('C:\\Users\\Ryan May\\Desktop\\Springboard\\us_job_market_discrimination.dta')

In [4]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [5]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## 1.)What test is appropiate? Does CLT Apply?
The goal of the test is to see if a applicant's race and whether or not they get a callback is related.  The poplulation standard deviation is unknown, so the 2-sample t-test will best fit the testing.  The dataset and since the race variable has just two variables ('b' and 'w'), the data is binominal population.  The binomial distribution can approximate a normal distrubition, so the CLT is not needed.

## 2.) What are the null and alternate hypotheses?
The null hypothesis is that white and black sounding names get the same call back rate.  In other words, the means of both populations will be the same.

The alternative hypothesis is that white and black sounding names will receive different call back rates.  The means will not be equal.

## 3.) What are the null and alternate hypotheses?

In [8]:
# separate the white sounding and black sounding names into two datasets
white_df=data[data['race']=='w']
black_df=data[data['race']=='b']

In [9]:
#numbers of resumes by race
n_w=len(white_df)
n_b=len(black_df)

In [15]:
#Find the rate of callbacks
rate_w=np.sum(white_df['call']/n_w)
rate_b=np.sum(black_df['call']/n_b)
print('Rate of Call Backs for Whites', rate_w)
print('Rate of Call Backs for Blacks', rate_b)
rate_diff=rate_w - rate_b
print('Difference Between Call Back Rates', rate_diff)

Rate of Call Backs for Whites 0.0965092
Rate of Call Backs for Blacks 0.0644764
Difference Between Call Back Rates 0.0320329


In [24]:
#finding the T-Score
t_stat, p=stats.ttest_ind(white_df['call'], black_df['call'],equal_var= False)
print('T-Stat', t_stat)
print('p-value', p)

T-Stat 4.11470529086
p-value 3.94294151365e-05


In [18]:
#Finding the Standard Error
s_error=np.sqrt(white_df['call'].var()/n_w + black_df['call'].var()/n_b)

In [19]:
#Finding the Margin of Error
margin_error=1.96*s_error
print('Margin of Error', margin_error)

Margin of Error 0.0152584173807


In [20]:
#Finding Confidence Interval
c_int= rate_diff +(np.array([-1,1])*margin_error)
print('Confidence Interval', c_int)

Confidence Interval [ 0.01677444  0.04729128]


In [23]:
#Finding p-value
p_val=stats.norm.cdf(-t_stat)*2
print('P-Value', p_val)

P-Value 3.87674401894e-05


The differences is rate of call backs is roughtly 3.2%, meaning white sounding names receive 3.2% more call backs.  The dataset shows that there is a difference but is it statistically significant?

The t statistic is 4.11 and the p value is 0.00039.  The high t value and very small p value show that the null hypothesis should be rejected.  This dataset shows that white sounding names do receive more call backs than black sounding names.

## 4.)Write a story describing the statistical significance in the context or the original problem.

Racial discrimination has created a two-tiered society in this country and the effects are being felt by those searching for a job.  A recent study tried to determine what factors (including race) affect how often applicants receive call backs.  

Analysis of this study show that white sounding names receive a call back 9.65 % of the time and black sound names receive 6.45 %.  To confirm that this batch of observations is not statisically random, a two sample t test was run.  The two sample t tst show that the rave of the applicant has a statisically siginificant effect on their rate of call backs.

## 5.) Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis shows that race is an important factor but not the only factor or even the most important factor.  I would examine the education level next and then the gaps in employment history.